java/client/src/main/java/glide/api/RedisClusterClient.java

```diff
@@ -47,28 +52,72 @@ public static CompletableFuture<RedisClusterClient> CreateClient(
 
     @Override
     public CompletableFuture<ClusterValue<Object>> customCommand(String[] args) {
-        Command command =
-                Command.builder().requestType(Command.RequestType.CUSTOM_COMMAND).arguments(args).build();
         // TODO if a command returns a map as a single value, ClusterValue misleads user
         return commandManager.submitNewCommand(
-                command, response -> ClusterValue.of(handleObjectResponse(response)));
+                RequestType.CUSTOM_COMMAND,
+                args,
+                Optional.empty(),
+                response -> ClusterValue.of(handleObjectResponse(response)));
     }
 
     @Override
     @SuppressWarnings("unchecked")
     public CompletableFuture<ClusterValue<Object>> customCommand(String[] args, Route route) {
-        Command command =
-                Command.builder()
-                        .requestType(Command.RequestType.CUSTOM_COMMAND)
-                        .arguments(args)
-                        .route(route)
-                        .build();
-
         return commandManager.submitNewCommand(
-                command,
+                RequestType.CUSTOM_COMMAND,
+                args,
+                Optional.ofNullable(route),
                 response ->
                         route.isSingleNodeRoute()
                                 ? ClusterValue.ofSingleValue(handleObjectResponse(response))
                                 : ClusterValue.ofMultiValue((Map<String, Object>) handleObjectResponse(response)));
```

Comment:
I believe you are not actually casting the keys in to strings with this (Map<String, Object>) , at runtime they will still remain as type Object not string. We may need to cast keys individually.

Reply:
I will test this and update the examples in the PR description.